subsys/bluetooth/mesh/beacon.c

```diff
@@ -785,11 +841,13 @@ void bt_mesh_beacon_enable(void)
 		bt_mesh_subnet_foreach(subnet_beacon_enable);
 	}
 
+	beacon_in_prog = false;
```

Comment:
If beacon sending is in progress, it is not required to be reset.
I guess the current implementation is not fully correct. It resets all accumulated data of ongoing process and is able to spam by beacons.

Reply:
My thought for why we need to reset is that if the beacon_send progression is interrupted, either by error or other interaction, the `sub_curr` will be in a unknown state. On the next iteration of beacon send, it will then skip whatever subnets that are "below" the `sub_curr`. If you have subnets 0, 1 and 2, a beacon send iteration is interupted while `sub_curr` is pointing on subnet 1, then in the next iteration beacon_send will only advertise subnets 1 and 2.